tests/unit/scm/test_scm.py

```diff
@@ -1,60 +1,61 @@
-from unittest import TestCase
+import pytest
 
-import mock
-
-from dvc.repo import Repo
 from dvc.repo.scm_context import scm_context
 from dvc.scm import NoSCM
 
 
-class TestScmContext(TestCase):
-    def setUp(self):
-        self.repo_mock = mock.Mock(spec=Repo)
-        self.scm_mock = mock.Mock(spec=NoSCM)
-        self.repo_mock.scm = self.scm_mock
+def test_scm_context(dvc, mocker):
+    scm = dvc.scm
+    m = mocker.patch.object(
+        scm, "track_file_changes", wraps=scm.track_file_changes
+    )
+
+    ret_val = 0
+    wrapped = scm_context(mocker.Mock(return_value=ret_val))
 
-    def test_should_successfully_perform_method(self):
-        method = mock.Mock()
-        wrapped = scm_context(method)
+    assert wrapped(dvc) == ret_val
+    m.assert_called_once_with(config=dvc.config)
 
-        self.repo_mock.configure_mock(config={})
-        wrapped(self.repo_mock)
 
-        self.assertEqual(1, method.call_count)
-        self.assertEqual(1, self.scm_mock.reset_ignores.call_count)
-        self.assertEqual(1, self.scm_mock.remind_to_track.call_count)
+def test_track_file_changes(mocker):
+    scm = mocker.Mock(NoSCM)
 
-        self.assertEqual(0, self.scm_mock.cleanup_ignores.call_count)
+    with NoSCM.track_file_changes(scm):
+        pass
 
-    def test_should_check_autostage(self):
-        method = mock.Mock()
-        wrapped = scm_context(method)
+    assert scm.reset_ignores.call_count == 1
```

Comment:
It's weird that these functions are mocked, but not tested at all separately. Made a minor improvement in the tests for now.